addons/zoom-slider/userscript.js

```diff
@@ -0,0 +1,79 @@
+export default async function ({ addon, msg }) {
+  const addSlider = async () => {
+    if (addon.tab.editorMode !== "editor" || addon.self.disabled) {
+      return;
+    }
+
+    const zoom = await addon.tab.waitForElement(".blocklyZoom");
+    const blockly = await addon.tab.traps.getBlockly();
+    window.blockly = blockly;
+
+    const foreignObject = document.createElementNS("http://www.w3.org/2000/svg", "foreignObject");
+    foreignObject.setAttribute("height", "500");
+    foreignObject.setAttribute("width", "120");
+    foreignObject.setAttribute("x", "-126");
+    foreignObject.setAttribute("y", "79");
+    const wrapperDiv = document.createElement("div");
+    wrapperDiv.classList.add("sa-zoom-slider-wrapper");
+    const slider = document.createElement("input");
+    slider.type = "range";
+    slider.min = blockly.mainWorkspace.options.zoomOptions.minScale;
+    slider.max = blockly.mainWorkspace.options.zoomOptions.maxScale;
+    slider.step = "0.01";
```

Comment:
Maybe I'm missing something but if you're already rounding the scale why not set the step to an integer?

Reply:
I don't think the scale's being rounded to an integer - that's only for displaying it as a percentage.